soccer/optimization/ParallelGradientAscent1D.cpp

```diff
@@ -0,0 +1,74 @@
+#include "ParallelGradientAscent1D.hpp"
+#include <math.h>
+#include <iostream>
+
+ParallelGradientAscent1D::ParallelGradientAscent1D(
+    ParallelGradient1DConfig* config)
+    : config(config) {
+    // Create list of GA1Ds
+    for (int i = 0; i < (config->GA1DConfig).size(); i++) {
```

Comment:
Not sure where you pushing for efficiency. But you could use reserve then transform here too if you want.